clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -20,13 +20,20 @@
 public class MockTime implements Time {
 
     private long nanos = 0;
+    private long autoTickMs = 0;
 
     public MockTime() {
         this.nanos = System.nanoTime();
     }
 
+    public MockTime(long autoTickMs) {
+        this.nanos = System.nanoTime();
+        this.autoTickMs = autoTickMs;
```

Comment:
Not sure I see why it would cause tests using nanoseconds() to fail, but I don't mind changing it. A separate idea I had is to implement a FastClock, which delegates to system time, but uses a constant multiplier to increase the clock rate.